src/grecha.ts

```diff
@@ -0,0 +1,93 @@
+import type { Filter } from "./filters";
+import type { Snapshot } from "./index";
+
+type Child = string | HTMLElement | Tag;
+
+/**
+ * Used to patch typescript conflict between class and className.
+ * Typscript merges interfaces so this works without modifying the library version of the interface.
+ * See the Notes section here.
+ * https://developer.mozilla.org/en-US/docs/Web/API/Element/className#notes
+ */
+// interface HTMLElement { class : string }
+
+export type Tag<T extends HTMLElement = HTMLElement> = T & {
```

Comment:
use generic functions to fix the massive `any` problem. This solution is almost perfect but there are issues with some errors in some places